include/picongpu/fields/laserProfiles/laserExpRampWithPrepulse.hpp

```diff
@@ -0,0 +1,188 @@
+/* Copyright 2013-2017 Axel Huebl, Heiko Burau, Rene Widera, Richard Pausch, Stefan Tietze
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+
+#pragma once
+
+#include "picongpu/simulation_defines.hpp"
+
+
+namespace picongpu
+{
+/** Wavepaket with spacial gaussian envelope and complicated longitudinal shape.
+ * Allows defining a prepulse and two regions of exponential preramp with inde-
+ * pendent slopes. The definition works by specifying three (t, intensity)-
+ * -points, where time is counted from the very beginning in SI and the inten-
+ * sity (yes, intensity, not amplitude) is given in multiples of the main peak.
+ *
+ * care! Problematic for few cycle laser pulses. Thought the rest is cloned 
+ * from laserWavepacket, the correctionFactor is not included (this was a 
+ * correction to the laser phase, which is necessary for very short pulses for
+ * the field to be physically consistent. Since I know the analytical solution
+ * only for the gaussian regime, and we have mostly exponential regimes here, 
+ * I dropped it.
+ */
+namespace laserExpRampWithPrepulse
+{
+    constexpr float_X laserTimeShift = laser::initPlaneY * CELL_HEIGHT /
+        SPEED_OF_LIGHT;
+    constexpr float_64 f = SPEED_OF_LIGHT / WAVE_LENGTH;
+    constexpr float_64 w = 2.0 * PI * f;
+    constexpr float_64 endUpramp = TIME_PEAKPULSE -
+        0.5 * LASER_NOFOCUS_CONSTANT;
+    constexpr float_64 startDownramp = TIME_PEAKPULSE +
+        0.5 * LASER_NOFOCUS_CONSTANT;
+
+    // helper functions, called from laserLongitudinal
+    HDINLINE float_X
+    gauss(float_64 t)
+    {
+        return math::exp(-0.25 * (t / PULSE_LENGTH) * (t / PULSE_LENGTH));
+    }
+
+    HDINLINE float_X
+    extrapolate(
```

Comment:
please use a more specific name: extrapolate *what* exactly?
This is not a simple linear interpolation but extrapolates ... something on the ramp?